engine/src/main/java/org/terasology/rendering/openvrprovider/OpenVRProvider.java

```diff
@@ -163,12 +189,87 @@ public void shutdown() {
         initialized = false;
     }
 
+    /**
+     * Query the VR library and update the VR state, gettable with getState(). This method should be called once per
+     * render cycle.
+     */
     public void updateState() {
         updatePose();
         pollControllers();
         pollInputEvents();
     }
 
+    /**
+     * Make the specified controller vibrate
+     * @param controller - the hand index, 0 for left and 1 for right.
+     * @param strength - the strength of the pulse.
+     */
+    public static void triggerHapticPulse(int controller, int strength) {
+        if (controllerDeviceIndex[controller] == -1) {
+            return;
+        }
+        vrSystem.TriggerHapticPulse.apply(controllerDeviceIndex[controller], 0, (short) strength);
+    }
+
+    public void submitFrame() {
+        for (int nEye = 0; nEye < 2; nEye++) {
+            vrCompositor.Submit.apply(
+                    nEye,
+                    texType[nEye], null,
+                    JOpenVRLibrary.EVRSubmitFlags.EVRSubmitFlags_Submit_Default);
+        }
+        if (vrCompositor.PostPresentHandoff != null) {
+            vrCompositor.PostPresentHandoff.apply();
+        }
+    }
+
+    /**
+     * Set the near clip. This will influence the projection matrix that comes from the
+     * vrProvider.getState().getProjectionMatrix(...) method.
+     * @param nearClipIn - the near clip to set.
+     */
+    public void setNearClip(float nearClipIn) {
+        nearClip = nearClipIn;
```

Comment:
I think the CheckStyle warning was not visible because "method not used" was blocking it. (I'm leaving this here because it could possibly be useful in the future, even if it's not currently used).